sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -927,7 +927,7 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
     }
     AlterTableDropPartitionCommand(
       visitTableIdentifier(ctx.tableIdentifier),
-      ctx.partitionSpec.asScala.map(visitNonOptionalPartitionSpec),
+      ctx.dropPartitionSpec().asScala.map(visitDropPartitionSpec),
```

Comment:
Can you update the comment?: https://github.com/apache/spark/blob/01c3dfab158d40653f8ce5d96f57220297545d5b/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala#L916

Reply:
mmmh, I am not sure how to update it. The only difference is that `specN` now can be any kind of filter instead of just `partitionColumn = value`. So it is actually the definition of specN which changed.